src/dra/7A4D0.c

```diff
@@ -1357,8 +1357,277 @@ void EntityPlayerBlinkWhite(Entity* self) {
+// Draws an outline around the player which grows or shrinks.
+// Outline can be several colors depending on the blueprint used.
+// Entity #31. Blueprints: 40, 57, 61
+// Many use cases. Known examples:
+// MP Refilled, blueprint 40, upperparams = 0
+// Cursed and trying to attack, blueprint 57, upperparams = 1
+// Agunea subweapon, blueprint 61, upperparams = 2
+// Dark Metamorphosis, blueprint 40, upperparams = 17
+// Sword warp spell, blueprint 61, upperparams = 20
+// Sword warp spell, blueprint 61, upperparams = 21
+// Soul steal, blueprint 40, upperparams = 22
+// Sword Brothers, blueprint 40, upperparams = 23
+
+void EntityPlayerOutline(Entity* self) {
+    s16* animFramePtr;
+    u8* spritesheetPtr;
+    s16 xOffset;
+    s16 yOffset;
+    s16 width;
+    Primitive* prim;
+    s16 spriteIdx;
+    s32 i;
+    s16 upperparams;
+    u8 spriteX;
+    s16* primData;
+    s16 xVar;
+    s16 yVar;
+    u8 four;
+    u8 one;
+    s16 height;
+    u8 spriteY;
+    s16 selfX;
+    s16 selfY;
+
+    if ((g_Player.unk0C & (PLAYER_STATUS_AXEARMOR | PLAYER_STATUS_UNK40000 |
+                           PLAYER_STATUS_STONE | PLAYER_STATUS_TRANSFORM)) ||
+        !(PLAYER.animCurFrame & 0x7FFF) || (!PLAYER.animSet) ||
+        ((PLAYER.step == Player_SpellHellfire) && (PLAYER.palette == 0x810D))) {
+        DestroyEntity(self);
+        return;
+    }
+    upperparams = (self->params & 0x7F00) >> 8;
+    self->posX.i.hi = PLAYER.posX.i.hi;
+    self->posY.i.hi = PLAYER.posY.i.hi;
+    self->facingLeft = PLAYER.facingLeft;
+    animFramePtr = D_800CF324[PLAYER.animCurFrame & 0x7FFF];
+    spriteIdx = *animFramePtr++;
+    spriteIdx &= 0x7FFF;
+    selfX = self->posX.i.hi;
+    selfY = self->posY.i.hi;
+    spritesheetPtr = ((u8**)SPRITESHEET_PTR)[spriteIdx];
+    four = 4;
+    one = 1;
+    spriteX = four + spritesheetPtr[0];
+    spriteY = one + spritesheetPtr[1];
+    width = spriteX - four;
+    height = spriteY - one;
+    xOffset = *animFramePtr++ + spritesheetPtr[2];
+    yOffset = *animFramePtr++ + spritesheetPtr[3];
+    self->rotZ = PLAYER.rotZ;
+    self->drawFlags |= (FLAG_DRAW_ROTX | FLAG_DRAW_ROTY);
+    primData = D_800AD9B8[upperparams];
+    switch (self->step) {
+    case 0: // Initialization
+        self->primIndex = AllocPrimitives(PRIM_GT4, 1);
+        if (self->primIndex == -1) {
+            DestroyEntity(self);
+            return;
+        }
+        self->flags = FLAG_UNK_04000000 | FLAG_HAS_PRIMS | FLAG_UNK_40000 |
+                      FLAG_UNK_20000 | FLAG_UNK_10000;
+        prim = &g_PrimBuf[self->primIndex];
+        // This is just not a for-loop, that's weird
+        for (i = 0; i < 1; i++) {
+            prim->tpage = 0x18;
+            prim->clut = primData[3]; // Always 259
+            prim->priority = PLAYER.zPriority + 2;
+            prim->drawMode =
+                primData[4] +
+                0x304; // Always 49, or DRAW_UNK_40 | DRAW_HIDE | DRAW_TRANSP
```

Comment:
Nit: `(DRAW_UNK_200 | DRAW_UNK_100 | DRAW_COLORS)`

Also I think the comment is meant for the previous line?

Reply:
This was all one line until the formatter broke it into 3, I'll clean this up, thanks!